src/silo/config/database_config.cpp

```diff
@@ -1,11 +1,16 @@
 #include "silo/config/database_config.h"
 
+#include <spdlog/spdlog.h>
+
 #include "silo/config/config_exception.h"
 
 namespace silo::config {
 
-DatabaseMetadataType toDatabaseMetadataType(std::string_view type) {
-   if (type == "string") {
+DatabaseMetadataType toDatabaseMetadataType(std::string_view type, bool generate_index) {
+   if (type == "string" && generate_index) {
+      return DatabaseMetadataType::INDEXED_STRING;
+   }
+   if (type == "string" && !generate_index) {
       return DatabaseMetadataType::STRING;
```

Comment:
So, do we want to hide the difference between `INDEXED_STRING` and `STRING` from the user? We could also simply make the user configure this directly instead of the additional field `generateIndex`.

Reply:
I guess this just makes it a lot simpler, afterall we prohibit non-indexing of pango lineages anyways. But maybe not hiding this and instead adding another helper function that deduces the used column type from the field `type` and `generate_index` is better afterall?